extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -35,13 +34,13 @@ export class NotebookUriHandler implements vscode.UriHandler {
 	}
 
 	private open(uri: vscode.Uri): Promise<void> {
-		const data = querystring.parse(uri.query);
+		const data = uri.query.toString().split('url=')[1];
```

Comment:
Wouldn't this fail if the query string doesn't have a url parameter? You'd end up with a length 1 array after the split.

Reply:
We should always have a url parameter so this shouldn't be a problem for valid URIs.

But you do need to account for url= being in folder/file names. So if you're going this route you should just find the first instance of that and split on that - not split the whole string.

Reply:
The azuredatastudio URI should follow standard URI standard. As a result, any link that a user clicks on should always have the 'url='.
`azuredatastudio://microsoft.notebook/open?url=`